source/extensions/quic_listeners/quiche/platform/quic_logging_impl.h

```diff
@@ -61,8 +61,10 @@
 
 #ifdef NDEBUG
 // Release build
-#define DCHECK(condition) QUIC_COMPILED_OUT_LOG()
 #define QUIC_COMPILED_OUT_LOG() QUIC_LOG_IMPL_INTERNAL(false, quic::NullLogStream().stream())
+#define DCHECK(condition)                                                                          \
```

Comment:
Do we actually need DCHECK to be defined? Can't we just use ASSERT unconditionally in code we're checking into the Envoy repo?

Reply:
I agree that we should use ASSERT in Envoy repo, but (D)CHECK is used in quiche code though.

Reply:
I'm still unclear how the build works here, and the doc should probably be included in, or referenced from the PR.
IIUC, quiche is an external library, which Envoy depends on. So how does Quiche get compiled if this macro has to be made available here, in Envoy code? Do we need to compile part of Quiche *after* compiling Envoy? Or do they all get built in one flow?

In particular, if you comment DCHECK out here, will this PR build?

Reply:
OK I think maybe some comments in BUILD files and here about the zig-zag dependencies between Envoy, Quiche, and the various layers of Quic defined in the Envoy codebase would really help.
BTW I don't see quic_logging_impl_lib as a BUILD target in the current code or in this PR. Did you mean quic_platform_base_impl_lib?

I assume at the end of this, everything just works building Envoy without having to worry about order; it should all be explicit in dependencies. 

It looks like:
 * quic_platform_base_impl_lib depends on a bunch of Envoy infrastructure, and that includes this file.
 * Some Quiche or other Quic infra depends on this
 * some other Envoy code must depend on Quiche or other Quic libraries.
Having these at the directory granularity, and having an explicit map will really help reason about the PR.

BTW I tried to patch your PR to compile it myself -- I wanted to see what fails without the DCHECK definitions, but the patch failed for me. You probably need a master merge.